backend/Routes/Controllers/auth/index.js

```diff
@@ -0,0 +1,278 @@
+              error: null,
+              results,
+            });
+          })
+          .catch((error) => {
+            return res.status(400).json({
+              success: false,
+              error,
+              results: null,
+            });
+          });
+    }
+);
+
+router.post(
+    '/update_password',
+    middleware.verifyUser.verifyAccessToken,
+    (req, res) => {
+      let validate = ajv.compile(updatePasswordSchema);
+      let valid = validate(req.body);
+      if (!valid) {
+        return res.status(400).json({
+          success: false,
+          error: validate.errors,
+          results: null,
+        });
+      }
+      auth
+          .updatePassword(req.body)
+          .then((results) => {
+            return res.status(200).json({
+              success: true,
+              error: null,
+              results,
+            });
+          })
+          .catch((error) => {
+            if (error === 'Password incorrect') {
+              return res.status(401).json({
+                success: false,
+                error,
+                results: null,
+              });
+            }
+            return res.status(400).json({
+              success: false,
+              error,
+              results: null,
+            });
+          });
+    }
+);
+
+router.post('/forgot_password', (req, res) => {
+  const validate = ajv.compile(forgotPasswordSchema);
+  const valid = validate(req.body);
+  if (!valid) {
+    return res.status(400).json({
+      success: false,
+      error: validate.errors,
+      results: null,
+    });
+  }
+  auth
+      .forgotPassword(req.body)
+      .then((results) => {
+        return res.status(200).json({
+          success: true,
+          error: null,
+          results,
+        });
+      })
+      .catch((error) => {
+        if (error === 'Email not linked to the username') {
+          return res.status(401).json({
+            success: false,
+            error,
+            results: null,
+          });
+        }
+        return res.status(400).json({
+          success: false,
+          error,
+          results: null,
+        });
+      });
+});
+
+router.post(
+    '/reset_password',
+    middleware.verifyUser.verifyAccessToken,
+    (req, res) => {
+      const validate = ajv.compile(resetPasswordSchema);
+      const valid = validate(req.body);
+      if (!valid) {
+        return res.status(400).json({
+          success: false,
+          error: validate.errors,
+          results: null,
+        });
+      }
+      auth
+          .resetPassword(req.body)
+          .then((results) => {
+            return res.status(200).json({
+              success: true,
+              error: null,
+              results,
+            });
+          })
+          .catch((error) => {
+            return res.status(400).json({
+              success: false,
+              error,
+              results: null,
+            });
+          });
+    }
+);
+
```

Comment:
At the end of every controller, could you use a function that returns 404 not found error, as if express is scanning upto here, then no URL has matched above property...

Reply:
ok